### ISSUE TITLE
Update fails when syncing product type with an attribute whose type was changed

### ISSUE THREAD

**Issue by kirmerzlikin** [commercetools/commercetools-sync-java#762]
**Describe the bug**
When syncing product type with an attribute whose type was changed sync library prepares invalid update which gets rejected by Commercetools platform.

**To Reproduce**
1. Create product type with attribute of type 'number'
```java
commercetoolsClient.executeBlocking(ProductTypeCreateCommand.of(
        ProductTypeDraftBuilder.of("test-type", "Test Type", "Test Type",
                        singletonList(AttributeDefinitionDraftBuilder.of(
                                NumberAttributeType.of(),
                                "test-attribute",
                                LocalizedString.ofEnglish("Test Attribute"),
                                false).build())
                ).build()));
```
2. Try to sync updated version of this product type with attribute type changed to 'text'
```java
ProductTypeDraftDsl updatedProductType =
        ProductTypeDraftBuilder.of("test-type", "Test Type", "Test Type",
                singletonList(AttributeDefinitionDraftBuilder.of(
                        StringAttributeType.of(),
                        "test-attribute",
                        LocalizedString.ofEnglish("Test Attribute"),
                        false).build())
        ).build();

ProductTypeSyncOptions productTypeSyncOptions = ProductTypeSyncOptionsBuilder.of(commercetoolsClient)
        .errorCallback((error, draft, productType, updateActions) -> {
            // getting error here
        }).build();
new ProductTypeSync(productTypeSyncOptions).sync(singletonList(updatedProductType));
```
3. Get product type sync error because Commmercetools platform responded with "400 Bad Request" and message "An attribute definition with name 'test-attribute' already exists on product type 'Test Type'"

**Expected behavior**
commercetools-sync library should not attempt to update product type if any of its attributes changed their types. It should instead throw `BuildUpdateActionException` indicating that this update is invalid.

**Additional context**
There is a [piece of code in `AttributeDefinitionsUpdateActionUtils`](https://github.com/commercetools/commercetools-sync-java/blob/master/src/main/java/com/commercetools/sync/producttypes/utils/AttributeDefinitionsUpdateActionUtils.java#L178) that appears to be handling this particular case by preparing two update actions - one removing existing attribute and one adding new one with updated type. But this approach doesn't work because Commercetools platform rejects such update.

**Comment by ahmetoz** [comment]
@kirmerzlikin thanks for reporting the issue and sorry for the late reply. I'm guessing sth changed on the API, will check the issue in detail and update the status asap.

**Comment by ahmetoz** [comment]
Hi @kirmerzlikin 
According to investigation the used workaround to change type is not allowed anymore, because removal of attributes from product types requires (async) background processing, the load can be significant.

Currently, we could not find another workaround to automate this in java-sync, we will discuss with the team for the approaches, one approach is like your suggestion and not creating those update actions and warning the user to do it separately. 

We are planning to do those changes in java-sync in next sprint time so in following 2 to 4 weeks.

Best Regards, Ahmet.

**Comment by kirmerzlikin** [comment]
Hi @ahmetoz 
Thanks for the explanation and update
I'll keep this issue open and we'll close it once corresponding changes are merged

**Comment by ahmetoz** [comment]
The issue is closed with release v8.0.0, for details please check: https://github.com/commercetools/commercetools-sync-java/blob/master/docs/RELEASE_NOTES.md#800---oct-1-2021

Hereby I'm closing the issue, please feel free to reply this issue in case of issue continues or re-open issue based on the request, thanks 🙏🏽
